src/odemis/driver/xt_client.py

```diff
@@ -2279,6 +2286,51 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
                 # know the user that the move is not complete.
                 if future._must_stop.is_set():
                     raise CancelledError()
+
+                # The stage is not moving anymore, but we still want to wait until the position has been updated
+                # TODO: base the timeout on the stage movement time estimation (est. *10)
+                timeout = time.time() + 10  # s
```

Comment:
a timeout is typically a certain amount of time to wait before cancelling something. `time.time()` returns the current UTC time, so this will be a very long timeout. On line 2313 it looks like you are using this more as an expected end time, so I would suggest to rename it to something like `expected_end_time`